pyoptsparse/pyParOpt/ParOpt.py

```diff
@@ -1,263 +1,24 @@
-                    """Evaluate the objective and constraint values"""
-                    fobj, fcon, fail = self.ptr._masterFunc(x[:], ["fobj", "fcon"])
-                    self.fobj = fobj
-                    return fail, fobj, -fcon
-
-                def evalObjConGradient(self, x, g, A):
-                    """Evaluate the objective and constraint gradients"""
-                    gobj, gcon, fail = self.ptr._masterFunc(x[:], ["gobj", "gcon"])
-                    g[:] = gobj[:]
-                    for i in range(self.m):
-                        A[i][:] = -gcon[i][:]
-                    return fail
-
-            optTime = MPI.Wtime()
-
-            # Optimize the problem
-            problem = Problem(self, n, m, xs, blx, bux)
-            optimizer = _ParOpt.Optimizer(problem, self.set_options)
-            optimizer.optimize()
-            x, z, zw, zl, zu = optimizer.getOptimizedPoint()
-
-            # Set the total opt time
-            optTime = MPI.Wtime() - optTime
-
-            # Get the obective function value
-            fobj = problem.fobj
-
-            if self.storeHistory:
-                self.metadata["endTime"] = datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")
-                self.metadata["optTime"] = optTime
-                self.hist.writeData("metadata", self.metadata)
-                self.hist.close()
-
-            # Create the optimization solution. Note that the signs on the multipliers
-            # are switch since ParOpt uses a formulation with c(x) >= 0, while pyOpt
-            # uses g(x) = -c(x) <= 0. Therefore the multipliers are reversed.
-            sol_inform = {"value": "", "text": ""}
-
-            # If number of constraints is zero, ParOpt returns z as None.
-            # Thus if there is no constraints, should pass an empty list
-            # to multipliers instead of z.
-            if z is not None:
-                sol = self._createSolution(optTime, sol_inform, fobj, x[:], multipliers=-z)
-            else:
-                sol = self._createSolution(optTime, sol_inform, fobj, x[:], multipliers=[])
-
-            # Indicate solution finished
-            self.optProb.comm.bcast(-1, root=0)
-        else:  # We are not on the root process so go into waiting loop:
-            self._waitLoop()
-            sol = None
-
-        # Communication solution and return
-        sol = self._communicateSolution(sol)
-
-        return sol
-
-    def _on_setOption(self, name, value):
-        """
-        Add the value to the set_options dictionary.
-        """
-        self.set_options[name] = value
+# First party modules
+from pyoptsparse.pyOpt_error import Error
+from pyoptsparse.pyOpt_optimizer import Optimizer
+
+try:
+    # External modules
+    from paropt.paropt_pyoptsparse import ParOptSparse as ParOpt
+except ImportError:
+
+    class ParOpt(Optimizer):
+        def __init__(self, raiseError=True, options={}):
+            name = "ParOpt"
```

Comment:
```suggestion
        def __init__(self, raiseError=True, options=None):
            options = options if options else {}
            name = "ParOpt"
```

Why? -> https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil

Reply:
Because this is the standard defined by the optimizer base class 

https://github.com/mdolab/pyoptsparse/blob/f66e9fbd6712dbb941794698cf87fc053c8a7986/pyoptsparse/pyOpt_optimizer.py#L34-L36

Feel free to open a PR making this fix for all optimizers